### PR TITLE
server connection with 2 players and all necessary data packages

### PR DESCRIPTION
**Pull request by pawlusha**
and server test, readme update

### CODE REVIEW

File: src/main/java/com/fourmusketeers/datapackages/DataPackageGenerator.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * Only use for tests or mock objects, not in game code!
+ *
+ * @author arek
+ * @since 26.07.17
+ */
+public class DataPackageGenerator {
+
+    public WhoStarts getWhoStarts(Boolean doIStart, String opponentName) {
```
Comment (arturAR) [inline src/main/java/com/fourmusketeers/datapackages/DataPackageGenerator.java]: 👍

File: src/main/java/com/fourmusketeers/datapackages/DataPackageGenerator.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        return new WhoStarts(doIStart, opponentName);
+    }
+
+    public InitialSetup getInitialSetup() {
+        List<Integer> list = new ArrayList<>();
```
Comment (arturAR) [inline src/main/java/com/fourmusketeers/datapackages/DataPackageGenerator.java]: What is this? Naming should be better.

File: src/main/java/com/fourmusketeers/datapackages/DataPackageGenerator.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    public GameTurnInfoToClient getGameTurnInfoToClient(Boolean canYplay) {
+        List<Integer> list = new ArrayList<>();
+        list.add(1);
+        return new GameTurnInfoToClient(list, false, 0, canYplay);
+    }
+
+    public EndGame getEndGame(Boolean isWin) {
+        return new EndGame(isWin);
+    }
```
Comment (patkkka) [inline src/main/java/com/fourmusketeers/datapackages/DataPackageGenerator.java]: All those methods aren't 'pure' getters. Consider changing 'get' to 'generate' or something else.

File: src/main/java/com/fourmusketeers/server/Connector.java [hunk 4]
```diff
@@ -23,12 +23,12 @@
         inputStream = new ObjectInputStream(socket.getInputStream());
     }
 
-    void sendOutput(String stringToSend) throws IOException {
-        outputStream.writeObject(stringToSend);
+    <T> void sendOutput(T dataToSend) throws IOException {
```
Comment (arturAR) [inline src/main/java/com/fourmusketeers/server/Connector.java]: Why "T? Why not your DataPackage interface?

File: src/test/java/com/fourmusketeers/server/ServerTest.java [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+    @Test
+    public void shouldCheckIfDataExists() throws IOException, ClassNotFoundException {
+        assertFalse(firstClientConnector.isThereAnyDataToRead());
+        server.getFirstClient().sendOutput(SOME_STRING);
+        assertTrue(firstClientConnector.isThereAnyDataToRead());
+        firstClientConnector.readInput();
+    }
+
+    @AfterClass(groups = {"sendingPackages"})
```
Comment (arturAR) [inline src/test/java/com/fourmusketeers/server/ServerTest.java]: 👍

File: src/main/java/com/fourmusketeers/server/Connector.java (hunk 4 above)
Comment (patkkka) [inline src/main/java/com/fourmusketeers/server/Connector.java]: 👍

File: src/test/java/com/fourmusketeers/server/ServerTest.java [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+        assertEquals(expected.isWin(), received.isWin());
+    }
+
+    @Test
+    public void shouldClientSendMessageToSecondClient() throws IOException, ClassNotFoundException {
+        firstClientConnector.sendOutput(SOME_STRING);
+        String message = server.getFirstClient().readInput();
+        server.getSecondClient().sendOutput(message.toUpperCase());
+
+        assertTrue(message.toUpperCase().equals(secondClientConnector.readInput()));
```
Comment (patkkka) [inline src/test/java/com/fourmusketeers/server/ServerTest.java]: Should be assertEquals :) beacuse it's operating on Strings.
Here is nice description what I mean:
[https://stackoverflow.com/questions/41974209/what-is-the-actual-difference-between-assertequals-vs-asserttrue-in-testng](url)

**Pull request merged**